scripts/lib-franklin.js

```diff
@@ -336,6 +336,31 @@ export async function translate(key, defaultText) {
   return defaultText;
 }
 
+export function decorateSupScript(string) {
```

Comment:
instead of creating separate arrays and then concatenate them, you could push the elements directly into a final array:

```
export function decorateSupScript(string, result = []) {
  if (!string) {
    return result;
  }

  const idx = string.search(/®|™/);

  if (idx !== -1) {
    result.push({
      type: 'span',
      textContent: string.substr(0, idx),
    },
    {
      type: 'sup',
      textContent: string.substr(idx, 1),
    });

    return decorateSupScript(string.substr(idx + 1), result);
  }

  result.push({
    type: 'span',
    textContent: string,
  });

  return result;
}
```

Reply:
Makes sense, PTAL again :-)